odl/operator/operator_utilities.py

```diff
@@ -0,0 +1,114 @@
+# Copyright 2014, 2015 The ODL development group
+#
+# This file is part of ODL.
+#
+# ODL is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ODL is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with ODL.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Usefull utility functions on discrete spaces (i.e., either Rn/Cn or
+discretized function spaces), for example obtaining a matrix representation of
+an operator. """
+
+# Imports for common Python 2/3 codebase
+from __future__ import print_function, division, absolute_import
+from future import standard_library
+standard_library.install_aliases()
+
+# External
+import numpy as np
+
+# Internal
+from odl.space.base_ntuples import FnBase
+from odl.set.pspace import ProductSpace
+
+
+def matrix_representation(op):
+    """Returns a matrix representation of a linear operator.
+
+    Parameters
+    ----------
+    op : :class:`~odl.Operator`
+        The linear operator of which one wants a matrix representation.
+
+    Returns
+    ----------
+    matrix : `numpy.ndarray`
+        The matrix representation of the operator.
+
+    Notes
+    ----------
+    The algorithm works by letting the operator act on all unit vectors, and
+    stacking the output as a matrix.
+
+    """
+
+    if not op.is_linear:
+        print('WARNING: The operator is not linear; cannot produce matrix',
+              'representation of it.')
+        return
+
+    if not (isinstance(op.domain, FnBase) or
+            isinstance(op.domain, ProductSpace)):
+        print('WARNING: The operator domain is not discrete or produc space;',
```

Comment:
Don't use commas in print statements like these, but apply auto-concatenation:
```python
>>> raise ValueError('you gave me {}, '  # No comma at the end!
                     'but I expected {}!'.format(1, 2))
Traceback (most recent call last):
...
ValueError: you gave me 1, but I expected 2!
```
will work nicely because the concatenation (triggered by writing 'a' 'b') either re-triggers or even precedes the `format` method, so that `format` is evaluated on the concatenated string. Compare:
```python
>>> print('{}' '{}'.format(1, 2))
12

>>> print('{}', '{}'.format(1, 2))
{} 1

>>> print('{}' + '{}'.format(1, 2))
{}1
```